wiki_articles/windows_msys2_gcc.md

```diff
@@ -0,0 +1,35 @@
+# How to install msys2 and gcc on windows
+In this guide, we install msys2 and gcc. There are some optional tools that we recommend
+installing, such as `clangd` and `clang-format`. Having an alternative compiler, `clang`
+is also recommended. Install location is assumed to be the default.
+
+<!-- inline -->
+## Msys2 and GCC Toolchain
+1. Visit [msys2.org](https://msys2.org) and follow their installation guide upto step 4.
+2. After installing, open the Msys2 Msys terminal environment on windows. It is an app.
+3. Type `pacman -Syu` and press Enter to update.
+4. Reopen Msys2 Msys, and now we install the entire toolchain group.
+5. Type `pacman -S mingw-w64-ucrt-x86_64-toolchain`. This will install all the necessary tools.
+
+<!-- inline -->
+## Clang (Optional)
+1. Type `pacman -S mingw-w64-ucrt-x86_64-clang` and press enter.
+2. (Optional) Install clang tools, `pacman -S mingw-w64-ucrt-x86_64-clang-tools-extra`.
+
+## Add to path
+1. Click on Search and type "Edit Environment Variables". Select `Path` and click Edit.
+2. In the new window, click on New and then type `C:\msys64\ucrt64\bin` and click Ok.
+3. Open powershell and type `gcc -v` to check for correct output. If not, try logging back in.
+
+## Extras
+Make is installed as `mingw32-make.exe`.
+You can also do this in `Msys2 Msys` to have it available as `make`:
+- `ln -sf /mnt/c/msys64/ucrt64/bin/mingw32-make.exe /mnt/c/msys64/ucrt64/bin/make.exe`
+
+## Notes
+- `clangd` is a language server, it provides code-completion, compiler errors and warnings etc.
+- `clang-format` as the name suggest is a C and C++ code formatter.
```

Comment:
I think I'd recommend not mentioning clangd or clang-format here just to keep this article short and straightforward. Arguably also leave clang for later.

Reply:
I believe that we should have the user install an alternate compiler, and having a different language server (clangd) is recommended as well.
This is because the vscode article builds up from a setup in this article.
I've seen how shitty microsoft intellisense extension for vscode is. VScode requires a c_cpp_properties.json (editor intellisense) and tasks.json (build) setup, which requires having to specify commands (compiler), and its arguments twice.

IMO It's just better to install clangd and cmake, and then have the build system generate your language server's config automatically. This is requires only specifying once in the build system file (CMakeLists.txt) and then everything builds up on top of that.